docs/modules/specifications/pages/decentral-nodes.adoc

```diff
@@ -0,0 +1,59 @@
+
+Also code a `TimedKeyValueStore<K,V>` interface that extends from `KeyValueStore<K,V>`, and provides a function `latestUpdate(key: K)` returning a timestamp of when the value of that key has been updated, as well as a function `hasUpdate(key: K, time: number)` that returns a boolean depending on whether the time is older or newer than when the value of that key has been updated.
+
+Combine both interfaces in a `TimedTypedKeyValueStore<M>` interface.
+
+Make the `MemoryStore` implementation extends from the combined interface, implementing the new methods of the `TimedKeyValueStore`, and make it take an `initialData` parameter in it's construction, so we can add data via the ComponentsJS configuration (n.b. the initial data simply take the current time as latest update).
+
+
+=== Scheduler daemon
+
+We'll need to check peer nodes of a SemCom Node server repetitively, but we want to abstract the repetition pattern from the task that needs to be repeated. Start by moving the `Daemon` class from `handlersjs-http` to a `models` directory in the `handlersjs-core` library, and make the necessary changes to imports and exports. 
+
+Then add a `daemons` directory, an implement a `Scheduler` that extends `Daemon`. This class can take an `interval` (in milliseconds) and a any function as `task`. When started, this daemon will set a timeout to wait for the given interval and then execute the task and set a new timeout. It should track the running timeout, so that when stopped it can cancel it.
+
+
+=== Synchronization service
+
+Code a `SyncService` class, which takes a `TypedKeyValueStore`, the key for a `storage` set therein, another key for a set of `peers` URLs and an optional a string containing an `endpoint` suffix. This service tracks the time of the last synchronization in a variable, and should contain an asynchronous `sync` function which runs through all URLs in the list of peers with the following steps:
+
+- Perform a GET request to the endpoint with an `If-Modified-Since` header containing the time of the last sync.
+- If the response is a 304, simply stores the current time as last sync.
+- If the response is a 200, add the response data to the storage set.
+
+
+=== JSON store handler
+
+Code a `JsonStoreHandler` extending `HttpHandler`, taking a `TimedTypedKeyValueStore` as well as a `data` key to access the data set therein. This handlers `handle` method should get the data from the store and return it as a stringified JSON response body. If the request contains an `If-Modified-Since` header, however, and the timestamp it contains is newer than the last update of the value in the store, it should instead send a 304 response without a body (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since[MDN]).
```

Comment:
```suggestion
Code a `JsonStoreHandler` extending `HttpHandler`, taking a `TimedTypedKeyValueStore` as well as a `data` key to access the data set therein. This handler's `handle` method should get the data from the store and return it as a stringified JSON response body. If the request contains an `If-Modified-Since` header, however, and the timestamp it contains is newer than the last update of the value in the store, it should instead send a 304 response without a body (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since[MDN]).
```

Reply:
just some typo's i found